autoload/ale/events.vim

```diff
@@ -138,6 +138,32 @@ function! ale#events#Init() abort
                 " not be changed here.
                 autocmd InsertLeave * if exists('*ale#engine#Cleanup') | call ale#cursor#EchoCursorWarning() | endif
             endif
+
+            function! AleShowCursorDetailCmd() abort
+                if exists('*ale#engine#Cleanup')
+                    call ale#preview#CloseIfTypeMatches('ale-preview')
+                    " Don't jump to the preview window during cursor movement.
+                    call ale#cursor#ShowCursorDetail({'stay_here': 1})
+                endif
+            endfunction
+
+            if g:ale_show_cursor_detail
+                if len(g:ale_cursor_detail_fileextensions)
+                    let ftStr=""
+
+                    for ft in g:ale_cursor_detail_fileextensions
```

Comment:
Install [Vint](https://github.com/Kuniwak/vint) and check your code with it. You should use `l:` for all local variables. Use `snake_case` for variable names, and `PascalCase` for function names.